### PR TITLE
Fix Zoom workflow so it can load non-event-data empty beam run

### PR DESCRIPTION
**Pull request by SimonHeybrock**
This should fix the issues seen in #71.

### CODE REVIEW

File: src/ess/polarization/zoom.py [hunk 1]
```diff
@@ -144,11 +160,11 @@ def ZoomTransmissionFractionWorkflow(runs: Sequence[str]) -> sl.Pipeline:
...
     workflow[MonitorSpectrumNumber[Incident]] = MonitorSpectrumNumber[Incident](3)
     workflow[MonitorSpectrumNumber[Transmission]] = MonitorSpectrumNumber[Transmission](
-        5
+        4
     )
-    workflow[NeXusMonitorName[Incident]] = NeXusMonitorName[Incident]('monitor3')
+    workflow[NeXusMonitorName[Incident]] = NeXusMonitorName[Incident]("monitor3")
     workflow[NeXusMonitorName[Transmission]] = NeXusMonitorName[Transmission](
-        'monitor5'
+        "monitor4"
```
Comment (SimonHeybrock) [inline src/ess/polarization/zoom.py:167]: @astellhorn Based on what you said on Slack this is wrong, and should be 2 and 3 instead of 3 and 4?

Comment (astellhorn) [reply src/ess/polarization/zoom.py:167]: Yes, its monitors called "monitor3" and "monitor4", but Diego (the instrument scientist) told be, it belongs to spectrum 2 ("monitor3") and spectrum 3 ("monitor4"), respectively.

Comment (astellhorn) [reply src/ess/polarization/zoom.py:167]: so we have to read out spectrum 2 and spectrum 3

Comment (SimonHeybrock) [reply src/ess/polarization/zoom.py:167]: In other words, this is correct? For example, the 2nd spectrum has spectrum number 3.

Comment (astellhorn) [reply src/ess/polarization/zoom.py:167]: It should be:
`    workflow[MonitorSpectrumNumber[Incident]] = MonitorSpectrumNumber[Incident](2)
    workflow[MonitorSpectrumNumber[Transmission]] = MonitorSpectrumNumber[Transmission](
        3
    )`

Comment (astellhorn) [reply src/ess/polarization/zoom.py:167]: I have tested. with that the line 
`raw_transmission = sans_workflow.compute(TransmissionFraction[SampleRun])`
(which gave the error before) is working

Comment (astellhorn) [reply src/ess/polarization/zoom.py:167]: But now I get a new error in the next line
`transmission_depolarized = raw_transmission['time', -1].copy()
transmission = raw_transmission['time', :-1].copy()
trans = sc.DataGroup(
    {f"{time:c}": transmission['time', time] for time in transmission.coords['time']}
)
trans[f'depolarized'] = transmission_depolarized
display(trans.plot())`:

---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[10], line 7
      3 trans = sc.DataGroup(
      4     {f"{time:c}": transmission['time', time] for time in transmission.coords['time']}
      5 )
      6 trans[f'depolarized'] = transmission_depolarized
----> 7 display(trans.plot())

File /opt/mamba/envs/mantid_env/lib/python3.10/site-packages/scipp/core/data_group.py:410, in DataGroup.plot(self, *args, **kwargs)
    407 def plot(self, *args, **kwargs):
    408     import plopp
--> 410     return plopp.plot(self, *args, **kwargs)

File /opt/mamba/envs/mantid_env/lib/python3.10/site-packages/plopp/plotting/plot.py:118, in plot(obj, aspect, cbar, coords, errorbars, figsize, grid, ignore_size, mask_color, norm, scale, title, vmin, vmax, autoscale, legend, **kwargs)
    116 ndim = ndims.pop()
    117 if ndim == 1:
--> 118     return figure1d(
    119         *nodes,
    120         errorbars=errorbars,
    121         mask_color=mask_color,
    122         legend=legend,
    123         **common_args,
    124     )
    125 elif ndim == 2:
    126     if len(nodes) > 1:

File /opt/mamba/envs/mantid_env/lib/python3.10/site-packages/plopp/graphics/figure.py:46, in figure1d(style, *args, **kwargs)
     43 if style == 'line':
     44     from .lineview import LineView
---> 46     return backends.figure1d(LineView, *args, **kwargs)
     48 raise ValueError(f'Unsupported style={style} for figure1d.')

File /opt/mamba/envs/mantid_env/lib/python3.10/site-packages/plopp/backends/manager.py:97, in BackendManager.figure1d(self, *args, **kwargs)
     95 except AttributeError:
     96     raise ValueError(f'Unsupported backend \'{self["2d"]}\' for figure1d.')
---> 97 return _figure1d(*args, **kwargs)

File /opt/mamba/envs/mantid_env/lib/python3.10/site-packages/plopp/backends/matplotlib/__init__.py:56, in MatplotlibBackend.figure1d(self, *args, **kwargs)
     51 def figure1d(self, *args, **kwargs):
     52     """
     53     See :class:`static.StaticFig` and :class:`interactive.InteractiveFig` for
     54     details.
     55     """
---> 56     return _make_figure(*args, **kwargs)

File /opt/mamba/envs/mantid_env/lib/python3.10/site-packages/plopp/backends/matplotlib/__init__.py:11, in _make_figure(*args, **kwargs)
      8 if is_interactive_backend():
      9     from .interactive import InteractiveFig
---> 11     return InteractiveFig(*args, **kwargs)
     12 else:
     13     from .static import StaticFig

File /opt/mamba/envs/mantid_env/lib/python3.10/site-packages/plopp/backends/matplotlib/interactive.py:16, in InteractiveFig.__init__(self, View, *args, **kwargs)
     15 def __init__(self, View, *args, **kwargs):
---> 16     self.__init_figure__(View, *args, **kwargs)
     17     self.toolbar = make_toolbar_canvas2d(
     18         canvas=self._view.canvas,
     19         colormapper=getattr(self._view, 'colormapper', None),
     20     )
     21     self.left_bar = VBar([self.toolbar])

File /opt/mamba/envs/mantid_env/lib/python3.10/site-packages/plopp/backends/matplotlib/figure.py:13, in Figure.__init_figure__(self, View, *args, **kwargs)
     12 def __init_figure__(self, View, *args, **kwargs):
---> 13     self._view = View(*args, **kwargs)
     14     self._args = args
     15     self._kwargs = kwargs

File /opt/mamba/envs/mantid_env/lib/python3.10/site-packages/plopp/graphics/lineview.py:113, in LineView.__init__(self, norm, vmin, vmax, autoscale, scale, errorbars, mask_color, aspect, grid, title, figsize, format, legend, *nodes, **kwargs)
    111 self.render()
    112 self._no_autoscale = False
--> 113 self.canvas.autoscale()
    114 self.canvas.finalize()

File /opt/mamba/envs/mantid_env/lib/python3.10/site-packages/plopp/backends/matplotlib/canvas.py:159, in Canvas.autoscale(self)
    152     line_x = sc.DataArray(data=sc.array(dims=['x'], values=line.get_xdata()))
    153     line_y = sc.DataArray(
    154         data=sc.array(dims=['x'], values=line.get_ydata()),
    155         masks={'mask': line_mask},
    156     )
    157     line_bbox = BoundingBox(
    158         **{**axis_bounds(('xmin', 'xmax'), line_x, self.xscale, pad=True)},
--> 159         **{**axis_bounds(('ymin', 'ymax'), line_y, self.yscale, pad=True)},
    160     )
    161     bbox = bbox.union(line_bbox)
    163 for c in self.ax.collections:

File /opt/mamba/envs/mantid_env/lib/python3.10/site-packages/plopp/backends/common.py:69, in axis_bounds(keys, x, scale, pad)
     47 def axis_bounds(
     48     keys: Tuple[str, str],
     49     x: sc.DataArray,
     50     scale: Literal['linear', 'log'],
     51     pad=False,
     52 ) -> Dict[str, float]:
     53     """
     54     Find sensible limits for an axis, depending on linear or log scale.
     55 
   (...)
     67         Whether to pad the limits.
     68     """
---> 69     values = fix_empty_range(find_limits(x, scale=scale, pad=pad))
     70     bounds = {k: v for k, v in zip(keys, (val.value for val in values))}
     71     return bounds

File /opt/mamba/envs/mantid_env/lib/python3.10/site-packages/plopp/core/limits.py:38, in find_limits(x, scale, pad)
     36 finite_inds = np.isfinite(v)
     37 if np.sum(finite_inds) == 0:
---> 38     raise ValueError("No finite values were found in array. Cannot compute limits.")
     39 finite_vals = v[finite_inds]
     40 finite_max = None

ValueError: No finite values were found in array. Cannot compute limits.

Comment (astellhorn) [reply src/ess/polarization/zoom.py:167]: > It should be: ` workflow[MonitorSpectrumNumber[Incident]] = MonitorSpectrumNumber[Incident](2) workflow[MonitorSpectrumNumber[Transmission]] = MonitorSpectrumNumber[Transmission]( 3 )`

Sorry, here I have a question: are these readouts of MonitorSpectrumNumber starting at a value 0?
I.e., if here I choose `MonitorSpectrumNumber[Incident](2) `, is it "spectrum: 2"?

Comment (astellhorn) [reply src/ess/polarization/zoom.py:167]: Screenshot 2024-08-08 at 16 30 09

Comment (SimonHeybrock) [reply src/ess/polarization/zoom.py:167]: What you see in the plot is a zero-based index, **not the spectrum number**. You have to check with the ISIS instrument scientists how they label their monitors, but I believe typically they start at 1.

Comment (SimonHeybrock) [reply src/ess/polarization/zoom.py:167]: > But now I get a new error in the next line `transmission_depolarized = raw_transmission['time', -1].copy() transmission = raw_transmission['time', :-1].copy() trans = sc.DataGroup( {f"{time:c}": transmission['time', time] for time in transmission.coords['time']} ) trans[f'depolarized'] = transmission_depolarized display(trans.plot())`:

Worked for me with the spectrum numbers as given in the PR.

Comment (SimonHeybrock) [reply src/ess/polarization/zoom.py:167]: Maybe this helps to explain how ISIS handles this: https://docs.mantidproject.org/nightly/tutorials/mantid_basic_course/loading_and_displaying_data/02_the_matrix_workspace.html#workspace-matrix-view-show-data

**Comment by SimonHeybrock** [comment]
@astellhorn Can you confirm that this is working and correct?

File: src/ess/polarization/zoom.py (hunk 1 above)
Comment (astellhorn) [reply src/ess/polarization/zoom.py:168]: > @astellhorn Can you confirm that this is working and correct?

Ah sorry as I am still working on the test data itself I forgot about this part of it. Yes, i can load it, it works perfectly! Please close this. 

For the testing of the workflow I am still wondering for some details why our results using our workflow are slightly different to the ones obained by the people at NIST, but I will open a new PR if needed.

**Review by astellhorn — APPROVED** [review]
This works perfectly now

**Pull request merged**